src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedFilingDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionFilingDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The recognition date or business start date-time string. */
+  protected get recognitionDateTime (): string {
+    if (this.isBenIaCorrectionFiling) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      }
+    }
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isAlterationFiling ||
+      this.isChangeRegFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectedFilingDate) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedFilingDate, true)
+      } else if (this.getBusinessFoundingDate) {
+        return this.apiToPacificDateLong(this.getBusinessFoundingDate)
+      }
+    }
+    return 'Unknown'
+  }
+
+  protected onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Called to add a new date. */
+  protected onOkClicked (date: string): void {
+    if (date) {
+      this.newCorrectedStartDate = date
+    }
+  }
+
+  protected onDoneClicked (): void {
+    if (this.newCorrectedStartDate) {
+      if (this.yyyyMmDdToPacificDate(this.newCorrectedStartDate, true) !==
+        this.apiToPacificDateLong(this.getBusinessFoundingDate)) {
+        this.setCorrectionFilingDate(this.newCorrectedStartDate)
+        this.isCorrected = true
+      } else {
+        this.newCorrectedStartDate = undefined
+      }
+    }
+    this.onEditMode = false
```

Comment:
So it the user clicks done but they haven't select a date, then the previous date is retained?

Reply:
Yes onOkClicked() sets the newCorrectedStartDate() which onDoneClicked() uses set the corrected filing date as long as it is not the same as the business founding date